crates/cranelift/src/gc/enabled.rs

```diff
@@ -1130,12 +1130,27 @@ impl FuncEnvironment<'_> {
         let base_offset = self.offsets.ptr.vmctx_gc_heap_base();
         let base_offset = i32::from(base_offset);
 
+        builder.ins().load(ptr_ty, flags, vmctx, base_offset)
+    }
+
+    /// Get the GC heap's bound.
+    fn get_gc_heap_bound(&mut self, builder: &mut FunctionBuilder) -> ir::Value {
+        let ptr_ty = self.pointer_type();
+        let flags = ir::MemFlags::trusted().with_readonly();
```

Comment:
Is the `with_readonly` here going to be accurate in the future? If I understand this right it's akin to `memory.size` which is readonly if `memory.grow` can't be used, but if we support for example more dynamic configurations of GC heaps in the future will this still be true?

Reply:
This code will all be refactored quite a bit for when we support backing GC heaps with linear memories and the static vs dynamic knobs and all that. This is really just for in the meantime, and to split out from some other stuff I am working on.